core/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/ThrottlingAllocationDecider.java

```diff
@@ -148,8 +153,12 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
                 }
                 int primaryNodeOutRecoveries = allocation.routingNodes().getOutgoingRecoveries(primaryShard.currentNodeId());
                 if (primaryNodeOutRecoveries >= concurrentOutgoingRecoveries) {
-                    return allocation.decision(THROTTLE, NAME, "too many outgoing shards are currently recovering [%d], limit: [%d]",
-                        primaryNodeOutRecoveries, concurrentOutgoingRecoveries);
+                    return allocation.decision(THROTTLE, NAME,
+                        "reached the limit of outgoing shard recoveries [%d], [%s=%d], [%s=%d]",
```

Comment:
It would be useful to know for which node this limit was reached. In case of incoming recoveries, it is the node where the shard is to be allocated to. In case of outgoing recoveries, however, the node is the one that has the primary. Let's add that node id here.